parlai/core/torch_ranker_agent.py

```diff
@@ -542,7 +540,19 @@ def _build_candidates(self, batch, source, mode):
                         label_vec = label_vec[0: cand_vecs[i].size(1)]
                     label_vec_pad[0: label_vec.size(0)] = label_vec
                     label_inds[i] = self._find_match(cand_vecs[i], label_vec_pad)
+                    if label_inds[i] == -1:
```

Comment:
You could do this without changing the semantics of `_find_match`:

```python
try:
    label_inds[i] = self._find_match(cand_vecs[i], label_vec_pad)
except RuntimeError:
    # got a bad batch.
    if opt.get('ignore_bad_candidates'):
        label_inds = None:
    else:
        raise
```

Reply:
i don't consider that better.
i actually think the way we are computing the loss isn't right, but it was too big a change for now. if we can track which ones we don't have labels for (set to -1) we can compute the loss correctly when labels are missing (they shouldn't be bad, they should just have 0 hits@1)